python/pyiceberg/catalog/base.py

```diff
@@ -57,7 +59,8 @@ def create_table(
         identifier: str | Identifier,
         schema: Schema,
         location: str | None = None,
-        partition_spec: PartitionSpec | None = None,
+        partition_spec: PartitionSpec = UNPARTITIONED_PARTITION_SPEC,
+        sort_order: SortOrder = UNSORTED_SORT_ORDER,
         properties: Properties | None = None,
```

Comment:
Should we similarly default `properties` to `{}`?

Reply:
I would like that, but that's actually a code smell in Python (or a quirk in the language). The default `{}` will be a reference to a single object, if you mutate that one, the next time the default value is being assigned, it will give the reference to the same object. More info here: https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/ So the recommended way is just to set it to null, and then do `properties or {}` in the code. This `{}` will then always initialize an empty dict.

Reply:
There's no immutable dict?

Reply:
Not by default, there is a package, or we can implement it ourselves.

Reply:
Maybe later, but this is fine for now.